views/ddox.docpage.dt

```diff
@@ -14,60 +16,79 @@ block ddox.title
 		- title = prefix ~ names.join("/") ~ " - multiple declarations";
 	
 block ddox.members
+	- bool multi_page = info.docGroups.length > 1;
 
-	- if (info.docGroups.length > 1)
-		ul
-			- foreach (dg; info.docGroups)
-				- auto itm = dg.members[0];
-				li #{itm.kindCaption} #{itm.nestedName}
+	- if (multi_page)
+		nav.page-nav
+			ul
+				- foreach (i, dg; info.docGroups)
+					- auto itm = dg.members[0];
+					li: a(href="\##{i}") #{dg.kinds.joiner("/")} #{itm.nestedName}
 
+	include ddox.inc.utils
 	include ddox.inc.composite
 	include ddox.inc.enum
 	include ddox.inc.function
 	include ddox.inc.template
 	include ddox.inc.variable
 
-	- foreach (dg; info.docGroups)
-		- auto item = cast(Declaration)dg.members[0];
-		- assert (item !is null, "Unknown entity type: "~dg.members[0].classinfo.toString());
+	- foreach (i, dg; info.docGroups)
+		- bool got_function = false;
+		- int hlevel = 2 + multi_page;
+		- auto kinds = dg.kinds;
```

Comment:
It's used once in the beginning and once in each `foreach` iteration. Usually the number of elements that `kinds` operates on will be quite limited, but it performs a GC allocation and sorting, so it isn't really cheap. Especially the GC allocation can be significant (it should really be eliminated completely, but as long as there are others this doesn't have the highest priority).

This should be more obvious from the name, though. I'll rename `kinds` to `collectKinds`.